### PR TITLE
Scheduler/evaluate time restriction

### PR DESCRIPTION
**Pull request by arejayelle**
Added Time restriction to fitness evaluation to the design. Added tests. Changed all dependent tests and files as necessary

### CODE REVIEW

File: skedge-backend/scheduler/test/test-semester-generation/test-rank-fitness.js [hunk 1]
```diff
@@ -1,151 +1,151 @@
...
-        // act
-        var fitnessRank = evaluateFitness(semester[0]);
-
-        // assert
-        expect(fitnessRank).to.satisfy(function(fitnessRank){
-            return fitnessRank == 1365;
-        });
-
-    });
+// var expect = require('chai').expect;
```
Comment (felixlapierre1) [inline skedge-backend/scheduler/test/test-semester-generation/test-rank-fitness.js]: Remove commented code

File: skedge-backend/scheduler/test/test-semester-generation/test-rank-generation.js [hunk 2]
```diff
@@ -1,113 +1,113 @@
-var expect = require('chai').expect;
-var rankGeneration = require("../../semester-generation/fitness.js").rankGeneration;
-var individual = require("../../semester-generation/individual.js");
+// var expect = require('chai').expect;
```
Comment (felixlapierre1) [inline skedge-backend/scheduler/test/test-semester-generation/test-rank-generation.js]: Remove commented code

File: skedge-backend/scheduler/test/test-semester-generation/test-rank-fitness.js (hunk 1 above)
Comment (arejayelle) [reply skedge-backend/scheduler/test/test-semester-generation/test-rank-fitness.js]: See new commit

File: skedge-backend/scheduler/test/test-semester-generation/test-rank-generation.js (hunk 2 above)
Comment (arejayelle) [reply skedge-backend/scheduler/test/test-semester-generation/test-rank-generation.js]: See new commit

**Pull request merged**